src/napari_builtins/io/_read.py

```diff
@@ -485,3 +486,78 @@ def napari_get_reader(
         return _csv_reader
 
     return _magic_imreader
+
+
+@contextmanager
+def _patch_viewer_new():
+    """Context manager to patch the viewer's new method."""
+    from napari.viewer import Viewer, current_viewer
+
+    original_new = Viewer.__new__
+    original_init = Viewer.__init__
+
+    def patched_init(self, *args, **kwargs):
+        Viewer.__init__ = original_init
+
+    def patched_new(cls, *args, **kwargs):
+        ndisplay = None
+        if len(kwargs) == 1 and 'ndisplay' in kwargs:
+            ndisplay = kwargs.pop('ndisplay')
+
+        if not kwargs and not args:
+            viewer = current_viewer()
+            if ndisplay is not None:
+                viewer.dims.ndisplay = ndisplay
+            if viewer is not None:
+                Viewer.__new__ = original_new
+                return viewer
+        Viewer.__init__ = original_init
+        return original_new(cls)
+
+    Viewer.__new__ = patched_new
+    Viewer.__init__ = patched_init
+    try:
+        yield
+    finally:
+        Viewer.__new__ = original_new
+        Viewer.__init__ = original_init
+
+
+def load_and_execute_python_code(path: str) -> list['LayerData']:
+    """Load and execute Python code from a file.
+
+    Parameters
+    ----------
+    path : str
+        Path to the Python file to be executed.
+    """
+    code = Path(path).read_text()
+    with _patch_viewer_new():
+        try:
+            exec(code)
```

Comment:
```suggestion
    viewer = current_viewer()
    code = Path(path).read_text()
    env = {}
    with _patch_viewer_new():
        try:
            exec(code, env)
            viewer.window._qt_viewer.add_to_console_backlog(env)
            globals().update(env)
```
This works! With `from napari.viewer import current_viewer`
Things get pushed to the console!
Maybe something like this can be used?